src/classes/BDI_Donations.cls

```diff
@@ -711,5 +767,108 @@ public class BDI_Donations {
+    * @param dataImportToReview the DI to determine if it is from Elevate
+    * @return result 
+    */
+    private Boolean isElevatePaymentDI(DataImport__c di) {
+        return di.Payment_Elevate_Id__c != null;
+    }
+
+
+    /*******************************************************************************************************
+    * @description Populates new and updates existing payments as needed for elevate related transactions.
+    * @param dataImportsById map of Data import records by their ids. 
+    */
+    private void processPaymentsForElevateDIs (DataImport__c[] dataImports) {
+
+        if (dataImports != null && dataImports.size() > 0) {
+            DataImport__c[] newPaymentDIs = new DataImport__c[]{};
     
+            for (DataImport__c di : dataImports) {
+
+                // For new opportunities, and thus new payments, add to a list for querying the existing oppt.
+                if (di.DonationImportStatus__c == System.Label.bdiCreated) {
+                    newPaymentDIs.add(di);
+                // For existing payments determine if any fields are populated before attempting update
+                } else if (di.PaymentImported__c != null 
+                    && BDI_DataImportService.anyFieldsPopulatedForObjectMapping(di,dataImportFieldToPaymentField)){
+                    paymentsByDataImportId.put(di.Id,new npe01__OppPayment__c(Id = di.PaymentImported__c));
+                }
+            }
+
+            //If there are DIs that need new Payments then construct
+            if (newPaymentDIs.size() > 0) {
+                constructNewPayments(newPaymentDIs);
+            }
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Queries Opportunities from DIs that have a DonationImported__c value.  This is primarily to 
+    * retreive Oppt field data for Elevate Payments to be constructed inside of the BDI code.
+    * @param disWithOpptForQuery Data Import records that have the DonationImported__c field populated
+    * @return opportunitiesById the opportunities returned by the query
+    */
+    private Map<Id, Opportunity> queryElevateOppts(DataImport__c[] disWithOpptForQuery) {
+        Set<Id> opptIdsForQuery = new Set<Id>();
+        Map<Id, Opportunity> resultOpptsById = new Map<Id,Opportunity>();
+
+        for (DataImport__c di : disWithOpptForQuery) {
+            opptIdsForQuery.add(di.DonationImported__c);
+        }
+
+        Set<String> fieldNamesForQuery = new Set<String>{'Id','StageName','CloseDate','Amount','IsClosed'};
+
+        // Add all the payment mappings to the list of fields to be queried since they will
+        // need to be copied to the payment.
+        for (npe01__Payment_Field_Mapping_Settings__c pfms : PMT_PaymentCreator.paymentMappings.values()) {
+            fieldNamesForQuery.add(pfms.npe01__Opportunity_Field__c);
```

Comment:
we should probably .toLowerCase() these fields and the initial set of fields in "fieldNamesForQuery" to make it case insensitive when adding into the set.

Reply:
Yeah, probably a good call, made the change.